### PR TITLE
jvmtiGetThreadInfo: releaseVMThread requires VM access

### PR DESCRIPTION
**Pull request by babsingh**
Re-organize code to acquire VM access before invoking releaseVMThread
in jvmtiGetThreadInfo.

Related: #15765
Fixes: #16250

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
@gacholio can you please review?

File: runtime/jvmti/jvmtiThread.c [hunk 1]
```diff
@@ -528,32 +528,39 @@ jvmtiGetThreadInfo(jvmtiEnv *env,
...
-							vm->vthreadGroupID = id;
+						if ((NULL != constants) && (NULL != id)) {
+							tempGroup = (*jniEnv)->GetStaticObjectField(jniEnv, constants, id);
+							Assert_JVMTI_notNull(tempGroup);
 						}
 
-						tempGroup = (*jniEnv)->GetStaticObjectField(jniEnv, constants, id);
-						Assert_JVMTI_notNull(tempGroup);
 						vm->internalVMFunctions->internalEnterVMFromJNI(currentThread);
+						/* VM access is required before invoking releaseVMThread. */
```
Comment (gacholio) [inline runtime/jvmti/jvmtiThread.c:559]: I think the comment belongs before the call to enter the VM.

Comment (babsingh) [reply runtime/jvmti/jvmtiThread.c:558]: Updated.

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk19

**Comment by gacholio** [comment]
jenkins compile win jdk8

**Pull request merged**